lib/transformer/caster/exporter.rb

```diff
@@ -0,0 +1,33 @@
+module Transformer
+  module Caster
+    class Exporter < Base
+      GraphMethodValue = Struct.new(:method, :key, :value)
```

Comment:
Maybe I'm misunderstanding, but it looks like the `GraphMethodValue` class doesn't really serve much purpose. Correct me if I'm wrong, but it seems instances are created in `exportable_attributes` and then are immediately used (and discarded) in `analyze`.

I think the class could be reduced to:

```ruby
class Exporter < Base
  def analyze
    each_exportable_attribute do |graph_method, value|
      # Assumes add_graph_value is (optionally) modified to take a
      # GraphValue instead of two keys.
      @template.add_graph_values(graph_method, value)
    end

    @template
  end

  private

  def each_exportable_attribute
    @dataset_cast.graph_methods.each do |key, value|
      if value && (opts = GraphMethods.all[key])
        yield(opts, value)
      end
    end
  end
end
```

Reply:
I [refactored it something more](https://github.com/quintel/transformer/pull/1/files#diff-9fa4a5bc097f3052ee70ed184a6f6981) thanks due to the validations for the `graph_methods`. 👍